File: app/src/main/java/pratamawijaya/moviesinfokotlin/app/AppModule.java
package pratamawijaya.moviesinfokotlin.app;

import android.app.Application;
import android.support.annotation.NonNull;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import pratamawijaya.moviesinfokotlin.data.executor.JobExecutor;
import pratamawijaya.moviesinfokotlin.domain.executor.PostExecutionThread;
import pratamawijaya.moviesinfokotlin.domain.executor.ThreadExecutor;
import pratamawijaya.moviesinfokotlin.domain.executor.UIThread;

/**
 * Created by pratama on 9/30/16.
 */

@Module @Singleton public class AppModule {
  Application application;

  public AppModule(Application application) {
    this.application = application;
  }

  @Provides @NonNull @Singleton public Application providesApplication() {
    return application;
  }

  @Provides @Singleton @NonNull
  public ThreadExecutor provideThreadExecutor(JobExecutor jobExecutor) {
    return jobExecutor;
  }

  @Provides @Singleton @NonNull
  public PostExecutionThread providePostExecutionThread(UIThread uiThread) {
    return uiThread;
  }
}

File: app/src/main/kotlin/pratamawijaya/moviesinfokotlin/app/AppModule.kt
package pratamawijaya.moviesinfokotlin.app

import android.app.Application
import android.support.annotation.NonNull
import dagger.Module
import dagger.Provides
import pratamawijaya.moviesinfokotlin.data.executor.JobExecutor
import pratamawijaya.moviesinfokotlin.domain.executor.ThreadExecutor
import javax.inject.Singleton

/**
 * Created by Pratama Nur Wijaya
 * Date : Oct - 10/11/16
 * Project Name : MoviesInfoKotlin
 */
@Module
@Singleton
class AppModule constructor(private val application: Application) {

  @Provides @NonNull @Singleton fun providesApplication(): Application {
    return application
  }

  @Provides
  @Singleton
  @NonNull
  fun provideThreadExecutor(jobExecutor: JobExecutor): ThreadExecutor {
    return jobExecutor
  }

}